ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm

```diff
@@ -58,4 +81,179 @@ - (void)testShouldIgnoreVnodeType {
     XCTAssertFalse(ShouldIgnoreVnodeType(static_cast<vtype>(1000), testVnode.get()));
 }
 
+- (void)testFileFlaggedInRoot {
+    bool fileFlaggedInRoot;
+    shared_ptr<mount> testMount = mount::Create();
+    shared_ptr<vnode> testVnode = vnode::Create(testMount, "/foo");
+    
+    testVnode->values.getattr = FileFlags_IsInVirtualizationRoot;
+    XCTAssertTrue(TryGetFileIsFlaggedAsInRoot(testVnode.get(), NULL, &fileFlaggedInRoot));
+    XCTAssertTrue(fileFlaggedInRoot);
+    
+    testVnode->values.getattr = FileFlags_IsEmpty;
+    XCTAssertTrue(TryGetFileIsFlaggedAsInRoot(testVnode.get(), NULL, &fileFlaggedInRoot));
+    XCTAssertFalse(fileFlaggedInRoot);
+    
+    testVnode->values.getattr = FileFlags_Invalid;
+    XCTAssertTrue(TryGetFileIsFlaggedAsInRoot(testVnode.get(), NULL, &fileFlaggedInRoot));
+    XCTAssertFalse(fileFlaggedInRoot);
+    
+    testVnode->values.getattr = 100;
+    XCTAssertTrue(TryGetFileIsFlaggedAsInRoot(testVnode.get(), NULL, &fileFlaggedInRoot));
+    XCTAssertFalse(fileFlaggedInRoot);
+
+    testVnode->errors.getattr = EBADF;
+    XCTAssertFalse(TryGetFileIsFlaggedAsInRoot(testVnode.get(), NULL, &fileFlaggedInRoot));
+}
+
+- (void)testShouldHandleVnodeOpEvent {
+    // In Parameters
+    shared_ptr<mount> testMount = mount::Create();
+    shared_ptr<vnode> testVnode = vnode::Create(testMount, "/foo");
+    testVnode->values.getattr = FileFlags_IsInVirtualizationRoot;
+    PerfTracer perfTracer;
+    vfs_context_t _Nonnull context = vfs_context_create(NULL);
+    kauth_action_t action = KAUTH_VNODE_READ_DATA;
+    
+    // Out Parameters
+    vtype vnodeType;
+    uint32_t vnodeFileFlags;
+    int pid;
+    char procname[MAXCOMLEN + 1] = "";
```

Comment:
This shadows the global variable of the same name, is that intentional?

Reply:
No it wasn't, fixing..